svir/dialogs/load_losses_by_asset_as_layer_dialog.py

```diff
@@ -0,0 +1,195 @@
+                feat.setGeometry(QgsGeometry.fromPoint(
+                    QgsPoint(row['lon'], row['lat'])))
+                feats.append(feat)
+            # NOTE: to download the whole data not grouped by location
+            # loss_type = self.loss_type_cbx.currentText()
+            # tdata = self.groupby(self.npz_file, rlz, loss_type, taxonomy)
+            # for row in self.dataset:
+            #     if (taxonomy != 'All' and row['taxonomy'] != taxonomy):
+            #         continue
+            #     # add a feature
+            #     feat = QgsFeature(self.layer.pendingFields())
+            #     for field_name_idx, field_name in enumerate(
+            #             orig_field_names):
+            #         if field_name in ['lon', 'lat']:
+            #             continue
+            #         if 'string' in self.dataset[field_name].dtype.name:
+            #             value = str(row[field_name_idx])
+            #         else:
+            #             value = float(row[field_name_idx])
+            #         feat.setAttribute(field_names[field_name_idx], value)
+            #     feat.setGeometry(QgsGeometry.fromPoint(
+            #         QgsPoint(row['lon'], row['lat'])))
+            #     feats.append(feat)
+            added_ok = self.layer.addFeatures(feats, makeSelected=False)
+            if not added_ok:
+                msg = 'There was a problem adding features to the layer.'
+                log_msg(msg, level='C', message_bar=self.iface.messageBar())
+
+    def load_from_npz(self):
+        for rlz in self.rlzs:
+            if (self.load_selected_only_ckb.isChecked()
+                    and rlz != self.rlz_cbx.currentText()):
+                continue
+            for taxonomy in self.taxonomies:
+                if (self.load_selected_only_ckb.isChecked()
+                        and taxonomy != self.taxonomy_cbx.currentText()):
+                    continue
+                for loss_type in self.loss_types:
+                    if (self.load_selected_only_ckb.isChecked()
+                            and loss_type != self.loss_type_cbx.currentText()):
+                        continue
+                    with WaitCursorManager(
+                            'Creating layer for realization "%s", '
+                            ' taxonomy "%s" and loss type "%s"...' % (
+                            rlz, taxonomy, loss_type), self.iface):
+                        self.build_layer(
+                            rlz, taxonomy=taxonomy, loss_type=loss_type)
+                        self.style_maps()
+                        # if self.output_type == 'loss_curves':
+                        #     self.style_curves()
+                        # elif self.output_type == 'loss_maps':
+                        #     self.style_maps()
+        if self.npz_file is not None:
+            self.npz_file.close()
+
+    def groupby(self, npz, rlz, loss_type, taxonomy='All'):
```

Comment:
Since this is a generic functionality I would convert it into a function (there is no dependency on `self`) and move it in some `util.py` module.